### PR TITLE
add repl examples for builtin functions and operators

### PR DESCRIPTION
**Pull request by fmoor**
- added examples where they were missing
- fixed broken examples (array indices)
- made existing examples more consistent
  * added missing semicolons
  * used `edgedb-repl` code-blocks as much as possible to show output instead of using comments to indicate output
- every operator in `doc/edgeql/funcop/set.rst` had its own heading. I removed the headings because they didn't seem to be consistent with the other files.

### CODE REVIEW

File: doc/edgeql/funcop/comparison.rst [hunk 1]
```diff
@@ -35,6 +45,11 @@ EdgeDB supports the following comparison operators:
     Works the same as regular :eql:op:`=<EQ>`, but also allows
     comparing ``{}``.  Two ``{}`` are considered equal.
 
+    .. code-block:: edgeql-repl
+
+        db> SELECT {1} ?= {1.0};
```
[the remark this answers is not in the document]
Comment (fmoor) [reply doc/edgeql/funcop/comparison.rst]: Sure thing

[the remark this answers is not in the document]
Comment (fmoor) [reply doc/edgeql/funcop/comparison.rst]: done

File: doc/edgeql/funcop/comparison.rst [hunk 2]
```diff
@@ -35,6 +45,21 @@ EdgeDB supports the following comparison operators:
...
+        {True}
+
+    .. code-block:: edgeql-repl
+
+        db> SELECT {1} ?= {};
+        {False}
+
+    .. code-block:: edgeql-repl
+
+        db> SELECT <int64>{} ?= {};  # the type of the empty set requires disambiguation here
```
Comment (1st1) [inline doc/edgeql/funcop/comparison.rst]: Please drop the comment. The line is too long (our coding style requires 80 chars max per line) and, while informative, it's not relevant.

**Comment by 1st1** [comment]
Squashed and merged. Thanks!

**Pull request merged**